pow/src/main/java/tech/pegasys/artemis/pow/event/Eth2Genesis.java

```diff
@@ -49,28 +53,47 @@ public long getTime() {
     return time;
   }
 
+  @Override
+  public void filterOutputFields(List<String> outputFields) {
+    this.outputFieldMap.put("eventType", "Eth2Genesis");
+    for (String field : outputFields) {
+      switch (field) {
+        case "deposit_root":
+          this.outputFieldMap.put("deposit_root", deposit_root.toHexString());
+          break;
+
+        case "deposit_count":
+          this.outputFieldMap.put("deposit_count", deposit_count);
+          break;
+
+        case "time":
+          this.outputFieldMap.put("time", time);
+          break;
+      }
+    }
+  }
+
   @Override
   public String toJSON() {
     Gson gson = new GsonBuilder().create();
     GsonBuilder gsonBuilder = new GsonBuilder();
-
-    JsonObject eth2Genesis = new JsonObject();
-    eth2Genesis.addProperty("eventType", "Eth2Genesis");
-    eth2Genesis.addProperty("deposit_root", deposit_root.toHexString());
-    eth2Genesis.addProperty("deposit_count", deposit_count);
-    eth2Genesis.addProperty("time", time);
-
+    String jsonString = gson.toJson(this.outputFieldMap);
+    JsonObject eth2Genesis = gson.fromJson(jsonString, JsonObject.class);
     Gson customGson = gsonBuilder.setPrettyPrinting().create();
     return customGson.toJson(eth2Genesis);
   }
 
   @Override
   public String toCSV() {
-    return null;
+    String csvOutputString = "";
+    for (Object obj : this.outputFieldMap.values()) {
+      csvOutputString += "'" + obj.toString() + "',";
+    }
+    return csvOutputString.substring(0, csvOutputString.length() - 1);
```

Comment:
Need a check to ensure that csvOutputString.length() > 0

Reply:
Done. please review.

Reply:
👍